integration/tcs.md

```diff
@@ -0,0 +1,148 @@
+
+### Updating Expected Observation End Time
+
+To update the estimated end time of an observation, a simple HTTP PATCH request to the `/api/observations/<id>` will update the observation end time. 
+
+```json
+{"end": "2022-01-01T00:45:00.000"}
+```
+
+This will be visible to authorized clients of the observation portal, including a scheduler, so that the new end time can be taken into account when scheduling future observations.
+### Setting Exposure Start Time (`exposures_start_at`)
+
+To set this, the TCS would send a HTTP PATCH request to the `/api/configurationstatus/<id>` with a JSON payload of the form:
+
+```json
+{"exposures_start_at": "2022-01-01T00:30:00.000"}
+```
+
+Once the observation portal detects this update, it will update the end time of the observation automatically, using the overheads defined in the Configuration Database.
+
+## Setting Configuration(s) as NOT_ATTEMPTED
+When executing an observation’s configurations, the terminal NOT_ATTEMPTED state may be set to indicate that the configuration wasn’t attempted by the TCS. This API feature is particularly useful when the TCS wants to indicate to the observation portal and/or scheduler that an observation should be re-scheduled because one or more of its configurations weren’t able to be attempted. This is in contrast to the FAILED state, which indicates that an observatory issue caused the execution of the configuration to fail.
+
+The portal treats observations with configuration states of NOT_ATTEMPTED as follows:
+
+**Observation State** | **Final Action by OCS**
+---------------------|----------------------------
+One or more configurations marked as NOT_ATTEMPTED | 1. If any configurations still PENDING observation is PENDING <br><br> 2. Else, if all configurations have reached terminal state, observation is FAILED and will be re-scheduled.
+All configurations marked as NOT_ATTEMPTED | Observation is marked as NOT_ATTEMPTED and will be re-scheduled.
+
+### Rescheduling an Observation Started Late or Unable to Complete In Time
+
+One important use-case for the NOT_ATTEMPTED state is when the TCS is able to determine that a scheduled observation has been started late (excessive slew time, acquisition time, etc.) and will not be able to be completed before the observation’s window expires. If this is determined early during the execution of an observation, then the TCS can mark all configurations as NOT_ATTEMPTED, the Observation Portal will then mark the observation as NOT_ATTEMPTED, and a scheduler can immediately re-schedule the observation.
```

Comment:
What you say is true. In practice the way this works for our site-software is if an Observation is not started within 10 minutes of the start-time we set its Configurations to NOT-ATTEMPTED. This usually happens because there is a technical or weather interruption.